src/admin/resolvers/mutations/ApprovedItem.auth.integration.ts

```diff
@@ -0,0 +1,592 @@
+import sinon from 'sinon';
+import { ReviewedCorpusItemEventType } from '../../../events/types';
+import {
+  CREATE_APPROVED_ITEM,
+  REJECT_APPROVED_ITEM,
+  UPDATE_APPROVED_ITEM,
+  UPDATE_APPROVED_ITEM_AUTHORS,
+  UPLOAD_APPROVED_ITEM_IMAGE,
+} from './sample-mutations.gql';
+import { expect } from 'chai';
+import { createReadStream, unlinkSync, writeFileSync } from 'fs';
+import { Upload } from 'graphql-upload';
+
+describe('mutations: ApprovedItem - authentication checks', () => {
+  const eventEmitter = new CuratedCorpusEventEmitter();
+
+  // a standard set of inputs for this mutation
+  const input: CreateApprovedItemInput = {
+    prospectId: '123-abc',
+    title: 'Find Out How I Cured My Docker In 2 Days',
+    url: 'https://test.com/docker',
+    excerpt: 'A short summary of what this story is about',
+    authors: [
+      { name: 'Mark Twain', sortOrder: 1 },
+      { name: 'Jane Austen', sortOrder: 2 },
+    ],
+    status: CuratedStatus.CORPUS,
+    imageUrl: 'https://test.com/image.png',
+    language: 'DE',
+    publisher: 'Convective Cloud',
+    topic: Topics.TECHNOLOGY,
+    source: CorpusItemSource.PROSPECT,
+    isCollection: false,
+    isTimeSensitive: true,
+    isSyndicated: false,
+  };
+
+  beforeAll(async () => {
+    await clearDb(db);
+  });
+
+  describe('createApprovedCorpusItem mutation', () => {
+    it('should succeed if user has access to one of scheduled surfaces', async () => {
+      // set up event tracking
+      const eventTracker = sinon.fake();
+      eventEmitter.on(ReviewedCorpusItemEventType.ADD_ITEM, eventTracker);
+
+      // Set up auth headers with access to a single Scheduled Surface
+      const headers = {
+        name: 'Test User',
+        username: 'test.user@test.com',
+        groups: `group1,group2,${MozillaAccessGroup.NEW_TAB_CURATOR_ENGB}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers, eventEmitter);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: CREATE_APPROVED_ITEM,
+        variables: { data: input },
+      });
+
+      expect(result.errors).to.be.undefined;
+      expect(result.data).not.to.be.null;
+
+      // Expect to see all the input data we supplied in the Approved Item
+      // returned by the mutation
+      expect(result.data?.createApprovedCorpusItem).to.deep.include(input);
+
+      // Check that the ADD_ITEM event was fired successfully:
+      // 1 - Event was fired once!
+      expect(eventTracker.callCount).to.equal(1);
+      // 2 - Event has the right type.
+      expect(await eventTracker.getCall(0).args[0].eventType).to.equal(
+        ReviewedCorpusItemEventType.ADD_ITEM
+      );
+      // 3- Event has the right entity passed to it.
+      expect(
+        await eventTracker.getCall(0).args[0].reviewedCorpusItem.externalId
+      ).to.equal(result.data?.createApprovedCorpusItem.externalId);
+
+      await server.stop();
```

Comment:
nit - we can get rid of all the `server.start()` and `server.stop()` calls.

Reply:
Awesome! They were put in initially because tests would hang on CircleCI, resulting in a failed check.